sktime/forecasting/compose/_autoregressive.py

```diff
@@ -0,0 +1,303 @@
+    ... )
+    >>> len_levels = len(y_test.index.names)
+    >>> y_test = y_test.groupby(level=list(range(len_levels - 1))).apply(
+    ...     lambda x: x.droplevel(list(range(len_levels - 1))).iloc[:-20]
+    ... )
+    >>>
+    >>> # create forecaster and wrapper
+    >>> forecaster = PytorchForecastingTFT(trainer_params={
+    ...     "max_epochs": 1,
+    ... }) # doctest: +SKIP
+    >>> wrapper = AutoRegressiveWrapper(
+    ...     forecaster=forecaster,
+    ...     horizon_length=3, # prediction_length for forecaster
+    ...     aggregate_method=np.mean,
+    ... ) # doctest: +SKIP
+    >>> wrapper.fit(y) # doctest: +SKIP
+    >>>
+    >>> # forecast multiple forecasting horizons with same trained model
+    >>>
+    >>> wrapper.predict(fh=[1, 5]) # doctest: +SKIP
+                                c2
+    h0   h1    time
+    h0_0 h1_0  2000-02-20  5.278717
+            2000-02-24  5.303032
+        h1_1  2000-02-20  5.106282
+            2000-02-24  5.119611
+        h1_10 2000-02-20  4.924220
+    ...                         ...
+    h0_4 h1_97 2000-02-24  4.815347
+        h1_98 2000-02-20  5.022863
+            2000-02-24  5.044283
+        h1_99 2000-02-20  4.714049
+            2000-02-24  4.768126
+
+    [2000 rows x 1 columns]
+    >>>
+    >>> wrapper.predict(fh=[8, 9, 10, 11, 12]) # doctest: +SKIP
+                                c2
+    h0   h1    time
+    h0_0 h1_0  2000-02-27  5.312222
+            2000-02-28  5.323535
+            2000-02-29  5.351599
+            2000-03-01  5.367970
+            2000-03-02  5.379033
+    ...                         ...
+    h0_4 h1_99 2000-02-27  4.754828
+            2000-02-28  4.766777
+            2000-02-29  4.872816
+            2000-03-01  4.886779
+            2000-03-02  4.896248
+
+    [5000 rows x 1 columns]
+    >>>
+    """
+
+    _tags = {
+        "scitype:y": "both",
+        "authors": ["geetu040"],
+        "maintainers": ["geetu040"],
+        "y_inner_mtype": [
+            "pd.DataFrame",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "X_inner_mtype": [
+            "pd.DataFrame",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "ignores-exogeneous-X": False,
+        "requires-fh-in-fit": False,
+        "capability:insample": False,
+        "capability:global_forecasting": True,
+    }
+
+    def __init__(self, forecaster, horizon_length, aggregate_method=None):
+        self.forecaster = forecaster
+        self.horizon_length = horizon_length
+        self.aggregate_method = (
+            aggregate_method if aggregate_method is not None else np.mean
+        )
+
+        super().__init__()
+
+    def _fit(self, y, X=None, fh=None):
+        # ignore the fh provided fit
+        _fh = ForecastingHorizon(range(1, self.horizon_length + 1))
+
+        self.forecaster.fit(y=y, X=X, fh=_fh)
+
+    def _predict(self, fh, X=None, y=None):
```

Comment:
Can the similar logic be used for `predict_interval` and `predict_quantiles` as well?

Reply:
I think that can be done but I'll have to look into that

Reply:
Right now we don't have any global forecaster that supports this, so this can be a future feature to add, I'll create an issue for this - just so we remember this in future.

Reply:
I think @Xinyu added probabilistic capabilities to pytorch ones, is it not the case? It's at least in progress, if not merged.

Reply:
It's in https://github.com/sktime/sktime/pull/6628.

Reply:
This PR currently implements auto regressive forecasting for global interfaces (`fit` and `predict`) but can be extended for `predict_quantiles` which the global forecasting base class doesnot support for now
1. wait for https://github.com/sktime/sktime/pull/6628 to be merged and then continue with `predict_quantile` in this PR for `AutoregressiveWrapper` (blocking)
2. continue with this PR and extend `AutoregressiveWrapper` for `predict_quantile` in another PR later (non-blocking)

So how do we proceed with this now, from the above 2 options?
@benHeid your suggestion is also needed here